### PR TITLE
mgr/dashboard: Verify fields on Configuration page

### PR DESCRIPTION
**Pull request by rafaelquint**
Fixes: https://tracker.ceph.com/issues/41193

Signed-off-by: Adam King <adking@redhat.com>
Signed-off-by: Rafael Quintero <rquinter@redhat.com>

### CODE REVIEW

**Comment by s0nea** [comment]
jenkins test make check

**Comment by Devp00l** [comment]
Sorry I have to say that, but your new tests can be formulated as simple unit tests as they are not integration tests.

Please add them as unit tests instead into `configuration.component.spec.ts`.

Thanks in advance.

**Comment by rafaelquint** [comment]
Added a couple additional tests

**Comment by p-na** [comment]
jenkins test make check

**Comment by adk3798** [comment]
Taking over this PR since @rafaelquint 's internship has ended. Most recent push is changing some of the tests over to unit tests as per recommendation of @Devp00l . I haven't written any dashboard unit tests until now so any constructive criticism is appreciated.

**Comment by adk3798** [comment]
jenkins test docs

**Comment by adk3798** [comment]
jenkins test dashboard

**Comment by adk3798** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.spec.ts [hunk 1]
```diff
@@ -35,4 +36,13 @@ describe('ConfigurationComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should check header text', () => {
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.spec.ts]: Thanks for adding the unit test. Here is a diff how you can greatly reduce this tests.

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.spec.ts
index 5d62513aa7..712c478dab 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.spec.ts
@@ -38,11 +38,8 @@ describe('ConfigurationComponent', () => {
   });
 
   it('should check header text', () => {
-    const dataTableHeader = fixture.debugElement.query(By.css('.datatable-header'));
-    expect(dataTableHeader.nativeElement.textContent).toContain('Name');
-    expect(dataTableHeader.nativeElement.textContent).toContain('Description');
-    expect(dataTableHeader.nativeElement.textContent).toContain('Current value');
-    expect(dataTableHeader.nativeElement.textContent).toContain('Default');
-    expect(dataTableHeader.nativeElement.textContent).toContain('Editable');
+    expect(fixture.debugElement.query(By.css('.datatable-header')).nativeElement.textContent).toBe(
+      ['Name', 'Description', 'Current value', 'Default', 'Editable'].join('')
+    );
   });
 });
```

**Comment by adk3798** [comment]
Rebased and fixed merge conflict

**Pull request merged**